bootstrap/lib/cache.js

```diff
@@ -37,31 +50,58 @@ const withCache = (execute, options) => {
   // Algorithm we use to derive entry key
   const hashOptions = {
     algorithm: 'sha1',
+
     encoding: 'hex',
     excludeKeys: (key) => options.ignoredKeys.includes(key),
   }
+  const _getKey = (data) => hash(data, hashOptions)
+  const _getCoalescingKey = (key) => hash(`${options.requestCoalescingGroup}--${key}`, hashOptions)
 
   const _getMaxAge = (data) => {
     if (!data || !data.data) return options.maxAge
     if (isNaN(data.data.maxAge)) return options.maxAge
     return data.data.maxAge || options.maxAge
   }
 
-  return (data, callback) => {
-    const key = hash(data, hashOptions)
+  const _getWithCoalescing = async (key, coalescingKey, retries = 5, ms = 100) => {
+    if (retries === 0) return null
+    const entry = cache.get(key)
+    if (entry) {
+      logger.debug(`Request coalescing: Success`)
+      return entry
+    }
+    const inFlight = cache.get(coalescingKey)
+    if (!inFlight) return null
```

Comment:
Is there a possible scenario where the key won't be set before multiple other instances in a burst request try to access the coalescing key? This can happen if the backing cache mechanism is not strongly consistent AFAIK

Reply:
Thanks for a great comment! Was thinking of adding a random delay here, just to scramble the timing of the request to avoid issues like this.

Reply:
That's definitely an option, another option would be to have the cache be local, and have the remote cache be used only for storing the computation results.

Reply:
The local cache for in-flight marks would suffer from the same issue as we have now because of AWS/GCP Lambda concurrency model. They basically start new micro containers for every concurrent request, so the local cache is empty most of the time.

Reply:
Added random delay in [1bcc985](https://github.com/smartcontractkit/external-adapters-js/pull/93/commits/1bcc98500e3963cb2858fc6cd7e77c8884d701e5).